server/index.js

```diff
@@ -3,41 +3,64 @@ var bodyParser = require('body-parser');
 var db = require('./db/mongo.js');
 var path = require('path');
 var app = express();
+var React = require('react');
+// var Application = require('../client/src/components/App.jsx');
+var reactServer = require('react-dom/server');
+
+// var HTMLtemplate = require(path.join(__dirname, '../client/src/template.js'));
 
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }))
-app.use(express.static(path.join(__dirname, '../client/dist')));
-// app.use(limitTransfer);
-// UNCOMMENT FOR ANGULAR
-// app.use(express.static(__dirname + '/../angular-client'));
-// app.use(express.static(__dirname + '/../node_modules'));
+app.use(express.static(path.join(__dirname, '../client/dist'))); // this line becomes useless as we serve the template 
+
+
 
-app.get('/:name', function(req, res) {
+app.get('/res/:name', function(req, res) {
+    res.header("X-Content-Type", "text/javascript");
+    res.sendFile(path.join(__dirname, '/../client/dist/index.html'))
+});
+
+app.get('/API/res/:name', function(req, res) {
     var q = req.params.name;
     db.res(q, (err, data) => {
+        res.header("Access-Control-Allow-Origin", "*");
         if (err) {
             res.sendStatus(505);
         } else {
+            data = data[0];
             res.send(JSON.stringify(data));
         }
     })
 
-})
+});
+
+
+app.listen(3001, function() {
+    console.log('listening on port 3001!');
+});
 
 
-// app.get('/api/users', function(req, res) {
-//     var data = JSON.parse(Object.keys(req.query)[0]);
-//     console.log(data.name);
-//     db.getAllUsers(data.name, (err, data) => {
+
+// app.get('/res/:name', function(req, res) {
```

Comment:
The Code Review Checklist says not to include lots of commented out code, so it's just something to consider as you make more progress.

Reply:
fixed :)